ipa-core/src/protocol/hybrid/oprf.rs

```diff
@@ -0,0 +1,294 @@
+                value: 7,
+            },
+        ];
+        let indices_to_compare = [(0, 2), (1, 3), (1, 4), (1, 5)];
+
+        let shares: [Vec<HybridReport<BA8, BA3>>; 3] = records.iter().cloned().share();
+
+        let indistinguishable_reports: [Vec<IndistinguishableHybridReport<BA8, BA3>>; 3] = shares
+            .iter()
+            .map(|s| s.into_iter().map(|r| r.clone().into()).collect::<Vec<_>>())
+            .collect::<Vec<_>>()
+            .try_into()
+            .expect("Expected exactly 3 elements");
+
+        let chunked_reports: [Vec<Vec<IndistinguishableHybridReport<BA8, BA3>>>; 3] =
+            indistinguishable_reports
+                .iter()
+                .map(|vec| {
+                    let mid = vec.len() / SHARDS;
+                    vec.chunks(mid)
+                        .map(|chunk| chunk.to_vec())
+                        .collect::<Vec<_>>()
+                })
+                .collect::<Vec<_>>()
+                .try_into()
+                .expect("Expected exactly 3 elements");
+
+        // #[allow(clippy::large_futures)]
+        // let results = chunked_reports
+        //     .into_iter()
+        //     .zip(contexts)
+        //     .map(|(reports_by_helper, helper_ctxs)| {
+        //         reports_by_helper
+        //             .into_iter()
+        //             .zip(helper_ctxs)
+        //             .map(|(reports, ctx)| async move {
+        //                 compute_prf_for_inputs(ctx, &reports).await.unwrap()
+        //             })
+        //             .collect::<Vec<_>>()
+        //     })
+        //     .collect::<Vec<_>>();
+
+        let mut results = Vec::new();
+        for (reports_by_helper, helper_ctxs) in chunked_reports.into_iter().zip(contexts) {
+            for (reports, ctx) in reports_by_helper.into_iter().zip(helper_ctxs) {
+                let result = async move { compute_prf_for_inputs(ctx, &reports).await };
+                results.push(result);
+            }
+        }
+
+        #[allow(clippy::large_futures)]
+        let results = futures::future::try_join_all(results).await.unwrap();
+
+        let results = results
+            .chunks(SHARDS)
+            .map(|chunk| chunk.into_iter().flatten().collect::<Vec<_>>())
+            .collect::<Vec<_>>();
+
+        let prfs = results
+            .iter()
+            .map(|helper_results| {
+                helper_results
+                    .into_iter()
+                    .map(|r| r.prf_of_match_key)
+                    .collect::<Vec<_>>()
+            })
+            .collect::<Vec<_>>();
+
+        // remove before merging
+        println!("{:?}", prfs);
+
+        // check to make sure each helper has the same PRF values
+        assert!(prfs.iter().all(|x| *x == prfs[0]));
+
+        // check to make sure the reports with the same match keys have the same prf values
+        for (i, j) in indices_to_compare {
+            assert_eq!(prfs[0][i], prfs[0][j]);
+        }
```

Comment:
this is currently failing for the case where match keys are the same, but the OPRF is computed on different shards.